From 64d56d58415ec01a80c8d48cc2777f19fdc7487b Mon Sep 17 00:00:00 2001
From: Binbin Zhou <zhoubinbin@loongson.cn>
Date: Sat, 12 Oct 2024 17:58:40 +0800
Subject: [PATCH 008/104] FROMGIT: ASoC: loongson: Fix build warning when
 !CONFIG_PCI

Fixes the below if kernel config disable PCI support:

sound/soc/loongson/loongson_i2s_pci.c:157:1: warning: data definition has no type or storage class
     157 | module_pci_driver(loongson_i2s_driver);
         | ^~~~~~~~~~~~~~~~~
sound/soc/loongson/loongson_i2s_pci.c:157:1: error: type defaults to 'int' in declaration of 'module_pci_driver' [-Wimplicit-int]
sound/soc/loongson/loongson_i2s_pci.c:157:1: error: parameter names (without types) in function declaration [-Wdeclaration-missing-parameter-type]
sound/soc/loongson/loongson_i2s_pci.c:149:26: warning: 'loongson_i2s_driver' defined but not used [-Wunused-variable]
     149 | static struct pci_driver loongson_i2s_driver = {
         |                          ^~~~~~~~~~~~~~~~~~~

Add the appropriate Kconfig dependency.

Fixes: ba4c5fad598c ("ASoC: loongson: Add I2S controller driver as platform device")
Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202410101647.o1lI37ou-lkp@intel.com/
Closes: https://lore.kernel.org/oe-kbuild-all/202410101439.Tfn5aT6i-lkp@intel.com/
Signed-off-by: Binbin Zhou <zhoubinbin@loongson.cn>
Link: https://patch.msgid.link/20241012095840.965087-1-zhoubinbin@loongson.cn
Signed-off-by: Mark Brown <broonie@kernel.org>

(cherry picked from commit 45b3605089b41b81ba36b231fbb97e3037a51beb
 https://git.kernel.org/pub/scm/linux/kernel/git/broonie/sound.git for-next)
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 sound/soc/loongson/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sound/soc/loongson/Kconfig b/sound/soc/loongson/Kconfig
index e641ae4156d9..2d8291c1443c 100644
--- a/sound/soc/loongson/Kconfig
+++ b/sound/soc/loongson/Kconfig
@@ -16,6 +16,7 @@ config SND_SOC_LOONGSON_CARD
 config SND_SOC_LOONGSON_I2S_PCI
 	tristate "Loongson I2S-PCI Device Driver"
 	depends on LOONGARCH || COMPILE_TEST
+	depends on PCI
 	select REGMAP_MMIO
 	help
 	  Say Y or M if you want to add support for I2S driver for
-- 
2.47.0

